ffi/CMakeLists.txt

```diff
@@ -42,37 +32,178 @@ else()
 endif()
 
 
-# Define our shared library
+# Capture the package format
+# LLVMLITE_PACKAGE_FORMAT the target package format, if set must be one of
+# 'wheel' or 'conda', this is used by the llvmlite maintainers in testing.
+
+# Keep in sync with config.cpp defines
+set(LLVMLITE_PACKAGE_FORMAT_CONDA "1")
+set(LLVMLITE_PACKAGE_FORMAT_WHEEL "2")
+
+string(TOLOWER "${LLVMLITE_PACKAGE_FORMAT}" lowercase_LLVMLITE_PACKAGE_FORMAT)
+if(lowercase_LLVMLITE_PACKAGE_FORMAT STREQUAL "conda" OR
+   lowercase_LLVMLITE_PACKAGE_FORMAT STREQUAL "wheel")
+    message(STATUS
+    "LLVMLITE_PACKAGE_FORMAT option is set, capturing this is a \
+'${lowercase_LLVMLITE_PACKAGE_FORMAT}' format build")
+    if(lowercase_LLVMLITE_PACKAGE_FORMAT STREQUAL "conda")
+        add_definitions(
+        -DLLVMLITE_PACKAGE_FORMAT=${LLVMLITE_PACKAGE_FORMAT_CONDA})
+    else()
+        add_definitions(
+        -DLLVMLITE_PACKAGE_FORMAT=${LLVMLITE_PACKAGE_FORMAT_WHEEL})
+    endif()
+    message(STATUS
+    "LLVMLITE_PACKAGE_FORMAT is ${lowercase_LLVMLITE_PACKAGE_FORMAT}")
+elseif("${lowercase_LLVMLITE_PACKAGE_FORMAT}" STREQUAL "")
+    # present but not set
+else()
+    message(FATAL_ERROR "Invalid value for package format: \
+'${LLVMLITE_PACKAGE_FORMAT}', expect one of 'conda' or 'wheel'.")
+endif()
+
+# Inherited from Makefile system, not tested on unsupported targets (BSD).
+if(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD"
+   OR ${CMAKE_SYSTEM_NAME} MATCHES "NetBSD")
+    check_cxx_compiler_flag(-flto HAVE_FLTO)
+    if(NOT HAVE_FLTO)
+        message(FATAL_ERROR "-flto flag is not supported by the compiler")
+    endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -flto")
+endif()
```

Comment:
Won't the `(UNIX AND NOT APPLE)` block below (https://github.com/numba/llvmlite/pull/1252/files#diff-f6e7eedf0c7c833db8a473a2be9560055fe3e31af19fad2334f7377c05644d7aR200) set `-flto` regardless? I think this is superfluous? (I also notice `Makefile.linux` had `-flto` in it too)